bin/git-bulk

```diff
@@ -14,15 +14,25 @@ allwsmode=false
 #
 usage() {
   echo 1>&2 "usage: git bulk [-g] ([-a]|[-w <ws-name>]) <git command>"
-  echo 1>&2 "       git bulk --addworkspace <ws-name> <ws-root-directory>"
+  echo 1>&2 "       git bulk --addworkspace <ws-name> <ws-root-directory> (--from <URL or file>)"
   echo 1>&2 "       git bulk --removeworkspace <ws-name>"
   echo 1>&2 "       git bulk --addcurrent <ws-name>"
   echo 1>&2 "       git bulk --purge"
   echo 1>&2 "       git bulk --listall"
 }
 
 # add another workspace to global git config
-function addworkspace { git config --global bulkworkspaces."$wsname" "$wsdir"; }
+function addworkspace { 
+	git config --global bulkworkspaces."$wsname" "$wsdir";
+	if [ ! -z $source ]; then
+		echo $source
+		if ! cd "$wsdir" 2>/dev/null; then echo "Path of workspace doesn't exist, make it first."; exit 1; fi
+		if [[ $source =~ http?(s):// ]]; then
+			git clone $source
+		elif [ -f $source ]; then while read line; do git clone $line; done < $file
```

Comment:
Please quote `$source` to avoid word splitting

Reply:
did for `$source` and `$file` as well